src/api.rs

```diff
@@ -1,5 +1,66 @@
+pub use std::convert::TryFrom;
 use std::error::Error;
 
+bitflags! {
+    pub struct Access: u32 {
+        const R = 0b00000001;
+        const W = 0b00000010;
+        const X = 0b00000100;
+        const NIL = 0b00000000;
+        const DUMMY = 0b11111111;
+        const RW = Self::R.bits | Self::W.bits;
+        const WX = Self::W.bits | Self::X.bits;
+        const RX = Self::R.bits | Self::X.bits;
+        const RWX = Self::R.bits | Self::W.bits | Self::X.bits;
+    }
+}
+
+impl TryFrom<Access> for u8 {
```

Comment:
No need for `TryFrom` here since there is no way this conversion can fail. I would simply implement the `Into` trait for `Access`.